### PR TITLE
Adding button action for midjourney 

### PR DESCRIPTION
**Pull request by tomakhil**

### CODE REVIEW

File: src/autobots/action/action_type/action_text2img/action_text2img_midjourney.py [hunk 1]
```diff
@@ -16,15 +16,15 @@ class Text2ImgRunModel(BaseModel):
...
+class ActionCreateText2ImgImagineMidJourney(ActionCreate):
+    type: ActionType = ActionType.text2img_imagine_midjourney_ai
     config: DiscordReqModel
 
 
 
 class ActionText2ImgMidjourney(IAction[DiscordReqModel, DiscordReqModel, DiscordReqModel, Text2ImgRunModel, DiscordJobsApiResponse]):
 
-    type = ActionType.text2img_midjourney_ai
+    type = ActionType.text2img_button_midjourney_ai
```
Comment (anuragjha) [inline src/autobots/action/action_type/action_text2img/action_text2img_midjourney.py:27]: Update action type

File: src/autobots/action/action_type/action_text2img/action_text2img_midjourney.py [hunk 2]
```diff
@@ -16,15 +16,15 @@ class Text2ImgRunModel(BaseModel):
                                   description="Text prompt with description of the things you want in the image to be generated.")
 
 
-class ActionCreateText2ImgMidJourney(ActionCreate):
-    type: ActionType = ActionType.text2img_midjourney_ai
+class ActionCreateText2ImgImagineMidJourney(ActionCreate):
```
Comment (anuragjha) [inline src/autobots/action/action_type/action_text2img/action_text2img_midjourney.py:19]: update filename to match classname

File: src/autobots/api/v1/api_action_type/text2img/text2img_midjourney/api_actions_text2img_midjourney.py [hunk 3]
```diff
@@ -20,15 +22,32 @@
...
         user_orm = UserORM(id=UUID(user_res.user.id))
-        action_create = ActionCreateText2ImgMidJourney(**action.model_dump())
+        action_create = ActionCreateText2ImgImagineMidJourney(**action.model_dump())
+        action_doc = await UserActions(user_orm, db).create_action(
+            ActionCreate(**action_create.model_dump())
+        )
+        return action_doc
+    except Exception as e:
+        logger.error(str(e))
+        raise HTTPException(500)
```
Comment (anuragjha) [inline src/autobots/api/v1/api_action_type/text2img/text2img_midjourney/api_actions_text2img_midjourney.py:40]: catch HTTPException and raise the same also

File: src/autobots/api/v1/api_action_type/text2img/text2img_midjourney/api_actions_text2img_midjourney.py [hunk 4]
```diff
@@ -56,3 +75,22 @@ async def run_action_text2img_midjourney(
...
+        user_orm = UserORM(id=UUID(user_res.user.id))
+        action = await UserActions(user=user_orm, db=db).get_action(action_id)
+        text2img = ActionText2ImgMidjourneyButton(
+            DiscordReqModel.model_validate(action.config)
+        )
+        resp = await text2img.run_action(action_input)
+        return resp
+    except Exception as e:
+        logger.error(str(e))
+        raise HTTPException(500)
```
Comment (anuragjha) [inline src/autobots/api/v1/api_action_type/text2img/text2img_midjourney/api_actions_text2img_midjourney.py:96]: and raise same if HttpException

**Comment by seoolin** [comment]
Download MidJourney
https://www.deliveryoka.com/application/midjourney/

**Pull request merged**